deploy/cdk/marble-web-kiosk-export-pipeline/lib/build-project.ts

```diff
@@ -0,0 +1,98 @@
+import cdk = require('@aws-cdk/core')
+import codebuild = require('@aws-cdk/aws-codebuild')
+import { BuildEnvironmentVariableType } from '@aws-cdk/aws-codebuild'
+import { Role } from '@aws-cdk/aws-iam'
+
+export enum BuildProjectType {
+  BUILD,
+  DEPLOY,
+}
+
+export interface BuildProjectProps {
+  projectType: BuildProjectType,
+  role: Role,
+  stage: String,
+  context?: { [key: string]: any },
+}
+
+const getContextCliArguments = (props: BuildProjectProps) => {
+  let output = ''
+  // if context is undefined, create an empty object so it doesn't crash
+  const context: { [key: string]: any } = props.context || {}
+  for (const key in context) {
+    if (key) {
+      output += `-c ${key}="${context[key]}"`
+    }
+  }
+  return output
+}
+
+const getPhases = (props: BuildProjectProps) => {
```

Comment:
I know this is how I separated build and deploy, but after our various discussions I don't know that it's really necessary. For instance, this creates a total of 4 CodeBuild projects - build test, build prod, deploy test, deploy prod. It stands to reason that you could shrink it down either to:

A) Build and Deploy test; Build and Deploy prod
B) Build test and prod; deploy test; deploy prod

The result should be the same regardless, but fewer CodeBuild projects means less work and a quicker pipeline, at least in theory. Of course, to do the latter option you would have to pass environment as a CLI context parameter to the `build.sh` script, instead of using an environment variable.

Reply:
Thanks for your suggestion.  In the interest of expediency, I'd prefer to not make these changes now, but rather incorporate them into my next creation of a pipeline.